### PR TITLE
[TF][NEURON][BUILD] - Add neuron monitor to be started.

### PR DESCRIPTION
**Pull request by aws-vrnatham**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
In the application container to know if inferences are using neuron, the neuron monitor can be used. With this change option is given with set of env variables that starts the neuron monitor which posts the metrics to the cloudwatch

### Tests run
Ran the test locally to make sure it works. Can't add a test case yet because the ec2 instances are started with a role that does not allow cloudwatch metrics to be posted.

**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron"/"graviton">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils/__init__.py [hunk 1]
```diff
@@ -47,7 +47,7 @@
 ]
 ECS_AML2_GPU_USWEST2 = "ami-09ef8c43fa060063d"
 ECS_AML2_CPU_USWEST2 = "ami-014a2e30da708ee8b"
-NEURON_AL2_DLAMI = "ami-092059396c7e51f52"
+NEURON_AL2_DLAMI = "ami-03c4cdc89eca4dbcb"
```
Comment (saimidu) [inline test/test_utils/__init__.py:50]: Is this a new publicly available AMI?

File: test/dlc_tests/eks/eks_manifest_templates/tensorflow/inference/single_node_neuron_inference.yaml [hunk 2]
```diff
@@ -56,14 +53,16 @@ spec:
           value: <MODEL_NAME>
         - name: MODEL_BASE_PATH
           value: <MODEL_BASE_PATH>
+        - name: AWS_LOG_LEVEL
+          value: "3"
         - name: AWS_REGION
-          value: us-west-2
+          value: us-east-1
```
[the remark this answers is not in the document]
Comment (saimidu) [reply test/dlc_tests/eks/eks_manifest_templates/tensorflow/inference/single_node_neuron_inference.yaml:59]: The TF Neuron EKS test seems to continue to use `us-west-2` as the region for the test. Can we sync on why this happens?

[the remark this answers is not in the document]
Comment (tejaschumbalkar) [reply test/dlc_tests/eks/eks_manifest_templates/tensorflow/inference/single_node_neuron_inference.yaml:59]: This should not affect the new EKS infra.

File: test/dlc_tests/eks/tensorflow/inference/test_eks_tensorflow_inference.py [hunk 3]
```diff
@@ -25,7 +25,7 @@ def test_eks_tensorflow_neuron_inference(tensorflow_inference, neuron_only):
 
     search_replace_dict = {
         "<MODEL_NAME>": model_name,
-        "<MODEL_BASE_PATH>": f"https://aws-dlc-sample-models.s3.amazonaws.com",
+        "<MODEL_BASE_PATH>": f"s3://aws-dlc-sample-models",
```
[the remark this answers is not in the document]
Comment (aws-vrnatham) [reply test/dlc_tests/eks/tensorflow/inference/test_eks_tensorflow_inference.py:28]: No. We need to compile the model using neuron-cc. This location has the compiled model of the publicly available model.

File: test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/conftest.py [hunk 4]
```diff
@@ -184,6 +184,12 @@ def skip_gpu_instance_restricted_regions(region, instance_type):
 @pytest.fixture(autouse=True)
 def skip_by_device_type(request, instance_type):
     is_gpu = instance_type.lstrip("ml.")[0] in ["g", "p"]
+    is_neuron = instance_type == 'ml.inf1.xlarge'
+
+    #If neuron run only tests marked as neuron
+    if (is_neuron  and not request.node.get_closest_marker("neuron_test")):
```
[the remark this answers is not in the document]
Comment (aws-vrnatham) [reply test/sagemaker_tests/tensorflow/inference/test/integration/sagemaker/conftest.py:190]: ok. Will take care of it.
